src/Interpreters/Context.cpp

```diff
@@ -781,8 +781,10 @@ Strings Context::getWarnings() const
     {
         if (setting.isValueChanged() && setting.isObsolete())
         {
-            common_warnings.emplace_back("Some obsolete setting is changed. "
-                                         "Check 'select * from system.settings where changed' and read the changelog.");
+            common_warnings.emplace_back(
+                "Obsolete setting `" + setting.getName()
```

Comment:
It will produce multiple warnings, one for each changed obsolete setting. Is it expected?

Reply:
I mean, maybe we can print one message with the list of such settings